test_runner/performance/test_logical_replication.py

```diff
@@ -149,12 +149,16 @@ def test_subscriber_lag(
                 check_pgbench_still_running(pub_workload, "pub")
                 check_pgbench_still_running(sub_workload, "sub")
 
-                with (
-                    psycopg2.connect(pub_connstr) as pub_conn,
-                    psycopg2.connect(sub_connstr) as sub_conn,
-                ):
-                    with pub_conn.cursor() as pub_cur, sub_conn.cursor() as sub_cur:
-                        lag = measure_logical_replication_lag(sub_cur, pub_cur)
+                pub_conn = psycopg2.connect(pub_connstr)
+                sub_conn = psycopg2.connect(sub_connstr)
+                pub_conn.autocommit = True
+                sub_conn.autocommit = True
+
+                with pub_conn.cursor() as pub_cur, sub_conn.cursor() as sub_cur:
+                    lag = measure_logical_replication_lag(sub_cur, pub_cur)
+
+                pub_conn.close()
+                sub_conn.close()
```

Comment:
Do I understand correctly that we want to add `conn.autocommit = True`?

If so, should it be just:
```suggestion
                with (
                    psycopg2.connect(pub_connstr) as pub_conn,
                    psycopg2.connect(sub_connstr) as sub_conn,
                ):
                    pub_conn.autocommit = True
                    sub_conn.autocommit = True
                    with pub_conn.cursor() as pub_cur, sub_conn.cursor() as sub_cur:
                        lag = measure_logical_replication_lag(sub_cur, pub_cur)
```

Reply:
No, see the second part of the PR description with the associated links.

Reply:
We can seemingly bring this pattern back for psycopg3.